harsh-vador/tic-tac-toe/script.js

```diff
@@ -0,0 +1,363 @@
+var grid = document.getElementById("grid");
+var msg = document.querySelector(".message");
+var chooser = document.querySelector("form");
+var mark;
+var cells;
+var moves = 0;
+var userName = document.querySelector(".name");
+var table = document.querySelector(".table");
+var submitBtn = document.querySelector(".submit_button");
+let origBoard;
+let huPlayer = "O";
+let aiPlayer = "X";
+
+const winCombos = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 4, 8],
+  [6, 4, 2],
+  [2, 5, 8],
+  [1, 4, 7],
+  [0, 3, 6],
+];
+
+const diff_cells = document.querySelectorAll(".cell");
+
+function setPlayer() {
+  mark = this.value;
+  msg.textContent = "";
+  userName = userName.value;
+  this.checked = true;
+}
+
+function gamelaunch() {
+  if (document.getElementById("username").value != "") {
+    if (mark == "X") buildGrid();
+    else if (mark == "O") startGame();
+    else alert("Select Level");
+  } else {
+    alert("Enter Name");
+  }
+}
+
+function playerMove() {
+  if (this.textContent == "") {
+    this.textContent = mark;
+    moves++;
+    const winner = checkRow();
+    switchMark();
+    setTimeout(() => !winner && computerMove(), 500);
+  }
+}
+
+function computerMove() {
+  var emptyCells = [];
+  var random;
+
+  cells.forEach(function (cell) {
+    if (cell.textContent == "") {
+      emptyCells.push(cell);
+    }
+  });
+  random = Math.ceil(Math.random() * emptyCells.length) - 1;
+  emptyCells[random].textContent = mark;
+  moves++;
+  if (emptyCells[random].textContent == undefined) alert("tie");
+  checkRow();
+  switchMark();
+}
+
+function switchMark() {
+  if (mark == "X") {
+    mark = "O";
+  } else {
+    mark = "X";
+  }
+}
+
+function winner(a, b, c) {
+  if (moves >= 9) msg.textContent = "Game Tie";
+  if (
+    mark == "X" &&
+    a.textContent == mark &&
+    b.textContent == mark &&
+    c.textContent == mark
+  ) {
+    msg.textContent = userName.value + " is the winner!";
+    a.classList.add("winner");
+    b.classList.add("winner");
+    c.classList.add("winner");
+    grid.classList.add("disabled");
+    userName.value = "";
+    return true;
+  } else if (
+    mark == "O" &&
+    a.textContent == mark &&
+    b.textContent == mark &&
+    c.textContent == mark
+  ) {
+    msg.textContent = "Computer Wins";
+    a.classList.add("winner");
+    b.classList.add("winner");
+    c.classList.add("winner");
+    grid.classList.add("disabled");
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function checkRow() {
+  if (
+    winner(
+      document.getElementById("c1"),
+      document.getElementById("c2"),
+      document.getElementById("c3")
+    )
+  )
+    return true;
+  else if (
+    winner(
+      document.getElementById("c4"),
+      document.getElementById("c5"),
+      document.getElementById("c6")
+    )
+  )
+    return true;
```

Comment:
returning true on every condition?

Reply:
Yes, but now made a change in the logic by using OR